### PR TITLE
TUG Etapas checkpoint 1st round of trials

### PR DESCRIPTION
**Pull request by fbrand-new**
Hi,

This PR contains all the changes that were effective during the first rounds of trials on 4/4/2023.

Notably

- Added offline data analysis with TUG phases automatic detection.
- Application is now manually launched. Docker compose used to set up yarprun servers.
- Skeleton duplication issues fixed.
- Out of bounds check is added.
- Various small fixes.

### CODE REVIEW

File: modules/skeletonRetriever/src/main.cpp [hunk 1]
```diff
@@ -693,10 +693,14 @@ class Retriever : public RFModule
     /****************************************************************/
     bool update_gaze_frame()
     {
-        if (Property* p=gazePort.read(false))
+        Property* p;
+        if (p=gazePort.read(false)) 
         {
-            if (Bottle* b=p->find("depth_rgb").asList())
+            if (Bottle* b=p->find("depth_center").asList())
```
Comment (pattacini) [inline modules/skeletonRetriever/src/main.cpp:699]: Any rationale for this change?
If I recall it right, `depth_center` is just a geometric frame, whereas the keypoints of the skeleton are expressed in the sensor frame, which should be `depth_rgb` by configuration.

**Review by pattacini — CHANGES_REQUESTED** [review]
Hi @fbrand-new 

Nice job with the tests!

I was the maintainer of this repo, but as discussed F2f w/ @randaz81 I'm no longer going to deal with this duty although I've remained particularly fond of the skeleton library plus the `skeletonRetriever` and the `skeletonViewer` module, given my authorship. Thus, I'll continue to maintain these components and I can also give help with them, of course.

Thereby, would you mind cleaning up the code to remove the debug lines and the leftovers. You may keep them by yourself in a dedicated branch or fork. Thanks in advance!

I've also dropped an inline comment about the gaze frame.

**Comment by fbrand-new** [comment]
Hi @pattacini, thanks for the comments! Sure, I will make sure to clean up as much as I can the mess that has been left around (sorry for that).

File: modules/skeletonRetriever/src/main.cpp (hunk 1 above)
Comment (fbrand-new) [reply modules/skeletonRetriever/src/main.cpp:699]: I may need to summon @randaz81 for this one. We noticed that the skeleton was not aligned in the viewer compared to what we saw in the real world so we tried this change, but if I recall correctly it did not have any impact on what we saw in the viewer and I forgot to clean it up

Comment (pattacini) [reply modules/skeletonRetriever/src/main.cpp:699]: I think this is related to #320. The pipeline assumes that you're using the internal Realsense and not that mounted on top of the head.

**Comment by pattacini** [comment]
> Hi @pattacini, thanks for the comments! Sure, I will make sure to clean up as much as I can the mess that has been left around (sorry for that).

Never mind, no problem!
I've asked just to clean up `skeletonRetriever`, whose changes can be fully reverted, judging from the PR.

File: modules/skeletonRetriever/src/main.cpp (hunk 1 above)
Comment (fbrand-new) [reply modules/skeletonRetriever/src/main.cpp:699]: Yes indeed. This is also the source of some other bugs that we have encountered in the last days (e.g. finish-line detected below the floor by line-detector). I am reverting the change in the PR.

**Review by pattacini — APPROVED** [review]
Thanks heaps @fbrand-new 👍🏻

**Pull request merged**